isis/src/base/objs/CropApp/CropApp.h

```diff
@@ -0,0 +1,97 @@
+#ifndef CropApp_h
+#define CropApp_h
+#include <QObject>
+#include <QString>
+#include <QThread>
+
+namespace Isis{
+
+
+class Buffer;
+class Cube;
+class LineManager;
+
+class CropApp: public QObject {
+
+
+    Q_OBJECT
+    public:
+
+    CropApp(QString &from, QString &to, int ssample = 1, int nsamples = -1, int sinc=1,
+             int sline =1, int nlines=-1, int linc=1, bool propspice=true);
+
+    public slots:
+    void start();
+    void handleResults();
+
+    signals:
+
+    void operate(QString &from, QString &to, int ssample = 1,int nsamples = -1,int sinc=1,
+                 int sline =1,int nlines=-1,int linc=1,bool propspice=true,Cube *cube=NULL);
+
+
+    private:
+
+    QThread cropThread;
+
+    QString m_fromCube;
+    QString m_toCube;
+    int m_ssample;
+    int m_nsamples;
+    int m_sinc;
+    int m_sline;
+    int m_nlines;
+    int m_linc;
+    int m_sband;
+    LineManager *m_in;
+    Cube * m_cube;
+    bool m_propspice;
+
+
+    };
+
+class cropper:public QObject{
+
+Q_OBJECT
+public:
+
+  static int m_sline;
+  static int m_linc;
+  static int m_sband;
+  static int m_ssample;
+  static int m_sinc;
+  static int m_osamples;
+  static int m_olines;
+  static int m_obands;
+  static Cube *m_cube;
+  static QString m_outputCubeName;
+  static bool m_propspice;
+  static LineManager *m_in;
```

Comment:
Why are all of these static? If there are ever multiple cropper objects working in parallel, they will conflict over the static variables. These should just be public member variables.